tendermint/src/host_functions.rs

```diff
@@ -0,0 +1,134 @@
+use crate::signature::Verifier;
```

Comment:
As a general thought, "host functions" that contains only a `CryptoProvider` trait still feels like YAGNI to me.

I would personally suggest putting it under a `crypto` module (which should probably have been created a long time ago, with all of the existing cryptography code in the `tendermint` crate factored into it).